### PR TITLE
Improve logging when missing proto reports

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java [hunk 1]
```diff
@@ -54,36 +54,43 @@ public AbstractConfiguration(Configuration configuration) {
...
       }
-      LOG.info("Analyzer working directory contains " + count + " .pb file(s)");
 
       return Optional.of(analyzerOutputDir);
     } catch (IOException e) {
       throw new IllegalStateException("Could not check for .pb files in " + analyzerOutputDir.toAbsolutePath(), e);
     }
   }
 
+  private Optional<Path> empty(String msg, boolean silent) {
```
Comment (SonarTech) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java]: MAJOR Make "empty" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java [hunk 2]
```diff
@@ -88,7 +90,7 @@ private boolean shouldExecuteOnProject(FileSystem fs) {
 
   private void executeInternal(SensorContext context) {
     Optional<Path> roslynReportPath = config.roslynReportPath();
-    Optional<Path> protobufReportsDirectory = config.protobufReportPath();
+    Optional<Path> protobufReportsDirectory = config.protobufReportPath(false);
```
Comment (Evangelink) [inline sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java]: Could you just let us know what this `false` means?

File: sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpConfigurationTest.java [hunk 3]
```diff
@@ -61,16 +61,16 @@ private void createRoslynOut() throws IOException {
   public void onlyRoslynReportPresent() throws IOException {
     createRoslynOut();
     config = new CSharpConfiguration(settings.asConfig());
-    assertThat(config.protobufReportPath()).isNotPresent();
+    assertThat(config.protobufReportPath(true)).isNotPresent();
```
Comment (Evangelink) [inline sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpConfigurationTest.java]: Could you just let us know what this `true` means?

File: sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java (hunk 2 above)
Comment (dbmeneses) [reply sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java]: ok, I'm refactoring it and creating two different methods

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java [hunk 4]
```diff
@@ -55,35 +55,53 @@ public AbstractConfiguration(Configuration configuration) {
...
```
Comment (SonarTech) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java]: MAJOR Make "empty" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpConfigurationTest.java [hunk 5]
```diff
@@ -61,16 +61,16 @@ private void createRoslynOut() throws IOException {
   public void onlyRoslynReportPresent() throws IOException {
     createRoslynOut();
     config = new CSharpConfiguration(settings.asConfig());
-    assertThat(config.protobufReportPath()).isNotPresent();
+    assertThat(config.protobufReportPathSilent()).isNotPresent();
```
Comment (Evangelink) [inline sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpConfigurationTest.java]: Thanks!

**Pull request merged**